OMCompiler/SimulationRuntime/c/simulation/solver/initialization/initialization.c

```diff
@@ -679,7 +679,7 @@ void initSample(DATA* data, threadData_t *threadData, double startTime, double s
     }
   }
 
-  if(stopTime < data->simulationInfo->nextSampleEvent) {
+  if(!isnan(data->simulationInfo->nextSampleEvent) && stopTime < data->simulationInfo->nextSampleEvent) {
```

Comment:
Isn't that the wrong logic, though? 
If the sample event is `NAN`, you'd want to enter the first/true branch "there are no sample-events" below, so this should be `isnan(...) || ... `, isn't it?

Reply:
I'm not sure. Currently, stepping through the unpatched code, a NaN value jumps to the `else'-branch. I only patched out the SIGFPE throw, and did not touch the control flow.